plugin/scratch_plugin.lua

```diff
@@ -4,25 +4,53 @@ if vim.g.loaded_scratch == 1 then
   return
 end
 vim.g.loaded_scratch = 1
+vim.g.os_sep = vim.g.os_sep or vim.fn.has("win32") and "\\" or "/"
 
 -- create any global command that does not depend on user setup
 -- usually it is better to define most commands/mappings in the setup function
 -- Be careful to not overuse this file!
 
 -- TODO: remove those requires
-local scratch_api = require("scratch.api")
 
-local scratch_main = require("scratch")
-scratch_main.setup()
+local base_path = vim.fn.stdpath("cache") .. vim.g.os_sep .. "scratch.nvim" .. vim.g.os_sep
+---@type Scratch.Actor
+local default_config = {
+  scratch_file_dir = base_path, -- where your scratch files will be put
+  filetypes = { "lua", "js", "py", "sh" }, -- you can simply put filetype here
+  win_config = {
+    relative = "editor", -- Assuming you want the floating window relative to the editor
+    row = 2,
+    col = 5,
+    width = vim.api.nvim_win_get_width(0) - 10, -- Get the screen width - row * col
+    --api_get_option("lines") - 5,
+    height = vim.api.nvim_win_get_height(0) - 5, -- Get the screen height - col
+    style = "minimal",
+    border = "single",
+    title = "",
+  },
+  file_picker = "fzflua",
+  filetype_details = {},
+  localKeys = {},
+  manual_text = "MANUAL_INPUT",
+}
+local Actor = require("scratch.actor")
+---@type Scratch.Actor
+vim.g.scratch_actor = vim.g.scratch_actor or setmetatable(default_config, Actor)
```

Comment:
I'm not very fond of set a metatable to a `config`, which to me, looks like turn a config into a object instance.

I would accept to init a `actor` with a `config` pass to it's constructor method.